spring-integration-redis/src/test/java/org/springframework/integration/redis/store/RedisMessageGroupStoreTests.java

```diff
@@ -453,6 +456,74 @@ public void testJsonSerialization() throws Exception {
 		assertEquals(errorMessage.getOriginalMessage(), ((ErrorMessage) errorMessageResult).getOriginalMessage());
 		assertEquals(errorMessage.getPayload().getMessage(),
 				((ErrorMessage) errorMessageResult).getPayload().getMessage());
+
+		Message<Foo> fooMessage = new GenericMessage<>(new Foo("foo"));
+		try {
+			store.addMessagesToGroup(1, fooMessage);
+			fail("SerializationException expected");
+		}
+		catch (Exception e) {
+			assertThat(e.getCause().getCause(), instanceOf(IllegalArgumentException.class));
+			assertThat(e.getMessage(),
+					containsString("The class with " +
+							"org.springframework.integration.redis.store.RedisMessageGroupStoreTests$Foo and name of " +
+							"org.springframework.integration.redis.store.RedisMessageGroupStoreTests$Foo " +
+							"is not in the trusted packages:"));
+		}
+
```

Comment:
Right, we take care here exactly about *deserialization*. 
The problem that `addMessagesToGroup()` calls `addMessage(message)` which has `getMessage(messageId)`, therefore performs deserialization as well.

Having that I think we should go ahead and optimize `addMessage(message)` from the `addMessagesToGroup()` to something like `void doAddMessage(message)`. And we should not have extra call to the Redis/Gemfire.